client/src/components/BlogPostForm.tsx

```diff
@@ -1,25 +1,34 @@
 import { AiOutlineLeft } from 'react-icons/ai';
-import { Link, useNavigate } from 'react-router-dom';
 import { FaLocationDot } from 'react-icons/fa6';
-import { Entry, Photo } from '../lib/types';
-import { FormEvent, useState } from 'react';
+import { PiSpinnerGap } from 'react-icons/pi';
+import { Link, useNavigate } from 'react-router-dom';
+import { FormEvent, useContext, useState } from 'react';
+import { Entry, Photo, UserContextValues } from '../lib/types';
 import UnsplashGallery from './UnsplashGallery';
+import UserContext from './UserContext';
 import {
   createEntry,
   searchUnsplash,
   updateEntry,
 } from '../lib/fetchFunctions';
-import { PiSpinnerGap } from 'react-icons/pi';
 
 type props = {
   entry: Entry | undefined;
 };
 
 export default function BlogPostForm({ entry }: props) {
   const navigate = useNavigate();
+  const context = useContext<UserContextValues>(UserContext);
```

Comment:
This is why I like custom hooks whenever I create Context. Isn't the following much nicer?
```
const { user } = useUser();
// user will be of type User | undefined
```